libpod/networking_common.go

```diff
@@ -481,6 +481,15 @@ func (c *Container) NetworkConnect(nameOrID, netName string, netOpts types.PerNe
 
 	netOpts.Aliases = append(netOpts.Aliases, getExtraNetworkAliases(c)...)
 
+	// check whether interface is to be named after the host interface
+	// name left unspecified
+	if netOpts.InterfaceName == "" && c.runtime.config.Containers.InterfaceName == "host" {
```

Comment:
this is only adding he logic to network connect but you have to do this for create as well.
https://github.com/containers/podman/blob/1fc34d8e2f94a67270ec6cb09fbcdac00f61538d/libpod/runtime_ctr.go#L272